### PR TITLE
docs(python): improve `n_unique` and `approx_n_unique` docs

### PR DESCRIPTION
**Pull request by alexander-beedie**
Added a note about `null` being considered a distinct value in its own right for the purposes of `n_unique`, and extended the examples a little to better illustrate the functionality.

### CODE REVIEW

File: py-polars/polars/expr/expr.py [hunk 1]
```diff
@@ -2955,18 +2955,25 @@ def n_unique(self) -> Self:
         """
         Count unique values.
 
+        Notes
+        -----
+        `null` is considered to be a unique value for the purposes of this count.
```
Comment (stinodego) [inline py-polars/polars/expr/expr.py:2960]: "This count" sounds weird, maybe this is better?

```suggestion
        `null` is considered to be a unique value for the purposes of this operation.
```

**Review by stinodego — COMMENTED** [review]
Good to have this clarified! It has surprised me in the past.

File: py-polars/polars/expr/expr.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/polars/expr/expr.py:2960]: Works for me ;)

**Comment by alexander-beedie** [comment]
> EDIT: Looks the `approx_n_unique` outcome is not stable - might want to ignore the result of that example.

Done! Added a 'skip' directive 👍

**Review by stinodego — APPROVED** [review]

**Pull request merged**